jetbrains-core/src/software/aws/toolkits/jetbrains/settings/SamExecutableDetector.kt

```diff
@@ -0,0 +1,70 @@
+// Copyright 2018 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+package software.aws.toolkits.jetbrains.settings
+
+import com.intellij.openapi.util.SystemInfo
+import com.intellij.openapi.util.text.StringUtil
+import java.io.File
+
+open class SamExecutableDetector {
+    private companion object {
+        val UNIX_PATHS = arrayOf("/usr/local/bin", "/usr/bin")
+        const val UNIX_EXECUTABLE = "sam"
+
+        val WINDOWS_PATHS = arrayOf("C:\\Program Files\\Amazon\\AWSSAMCLI\\bin", "C:\\Program Files (x86)\\Amazon\\AWSSAMCLI\\bin")
+        val WINDOWS_EXECUTABLES = arrayOf("sam.cmd", "sam.exe")
+
+        const val PATH_ENV = "PATH"
+    }
+
+    fun detect(): String? = if (SystemInfo.isWindows) {
+        detectForWindows()
+    } else {
+        detectForUnix()
+    }
+
+    private fun detectForWindows(): String? {
+        for (path in WINDOWS_PATHS) {
```

Comment:
why the classic for loops and not a `foreach` ?

Reply:
Replaced